### PR TITLE
Revert "xorg.xorgserver: 1.20.13 -> 21.1.3"

### LINKED ISSUE #170856: Nouveau segfault when launching X11 Display Manager
### Describe the bug
I have a Dell XPS 9550, which has both an Intel I7 and Nvidia 950M. When I do a fresh NixOS install (using NixOS-unstable), by default lightdm and sddm both cause a segmentation fault that seems to be related to nouveau. Disabling the nouveau driver with `modprobe -r nouveau` allows me to restart the display manager and launch a desktop environment. This happend with both SDDM and KDE Plasma, and Light with Pantheon. Log: https://gist.github.com/vlinkz/62e48c55793402d98746b92ad50ccea8
```
[     9.341] (==) NOUVEAU(G0): Backing store enabled
[     9.341] (==) NOUVEAU(G0): Silken mouse enabled
[     9.341] (II) NOUVEAU(G0): [XvMC] Associated with Nouveau GeForce 8/9 Textured Video.
[     9.341] (II) NOUVEAU(G0): [XvMC] Extension initialized.
[     9.341] (==) NOUVEAU(G0): DPMS enabled
[     9.341] (EE) 
[     9.341] (EE) Backtrace:
[     9.341] (EE) 0: /nix/store/bnic4bpn6x1jy2j3glnzdw54km94p9xa-xorg-server-21.1.3/bin/X (OsSigHandler+0x25) [0x5b4065]
[     9.342] (EE) 1: /nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/libc.so.6 (__sigaction+0x40) [0x7fad63270100]
[     9.342] (EE) 2: /nix/store/bnic4bpn6x1jy2j3glnzdw54km94p9xa-xorg-server-21.1.3/bin/X (InitOutput+0x4e4) [0x487744]
[     9.342] (EE) 3: /nix/store/bnic4bpn6x1jy2j3glnzdw54km94p9xa-xorg-server-21.1.3/bin/X (dix_main+0x194) [0x448694]
[     9.343] (EE) 4: /nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/libc.so.6 (__libc_start_call_main+0x67) [0x7fad6325c1d7]
[     9.343] (EE) 5: /nix/store/ayrsyv7npr0lcbann4k9lxr19x813f0z-glibc-2.34-115/lib/libc.so.6 (__libc_start_main+0x87) [0x7fad6325c297]
[     9.343] (EE) 6: /nix/store/bnic4bpn6x1jy2j3glnzdw54km94p9xa-xorg-server-21.1.3/bin/X (_start+0x21) [0x431cd1]
[     9.343] (EE) 
[     9.343] (EE) Segmentation fault at address 0x10
[     9.343] (EE) 
Fatal server error:
[     9.343] (EE) Caught signal 11 (Segmentation fault). Server aborting
[     9.343] (EE) 
[     9.343] (EE) 
```
### Steps To Reproduce
1. Install NixOS unstable with all video driver options left at defaults
2. Enable an X11 display manager such as lightdm
3. Try to start display manager

### Expected behavior
Display manager starting

### Additional context
- Downgrading to NixOS 21.11 fixed the issue. Output log: https://gist.github.com/vlinkz/a7db75c7f0668f92369d0680669e747b
- Adding `"intel"` to `xserver.videoDrivers` also works, but seemed to be disabled by default in https://github.com/NixOS/nixpkgs/issues/55520 and https://github.com/NixOS/nixpkgs/pull/55583
- Setting `boot.kernelParams = [ "nouveau.modeset=0" ]` makes it work

### Notify maintainers

Not sure who
### Metadata
22.05pre372961.e10da1c7f54 - Not working
21.11.337266.feea25c5865 - Working

### PR DESCRIPTION
**Pull request by vcunat**
This reverts commit 3312352596a2c5c6bf6775ee2f90c4eca0e39223.
Fixes #170856: issues of some users.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
According to testing on the issue, reverting the driver bumps from PR #153994 isn't needed to fix the issues.  (There are no newer releases for -ati and -nouveau.)

However, there's a newer release of xorg-server-1.20.14, but the affected people haven't tested it (yet), so I can't be confident that it didn't backport the regression.  (There is some modesetting change.)   You can try https://github.com/vcunat/nixpkgs/commits/p/xorg-server_bump

**Comment by vcunat** [comment]
We've been using 21.1.3 for a very long time and I don't recall people reporting problems, so I'd hope it would be OK even without doing more testing beforehand.

**Comment by jakobkukla** [comment]
> However, there's a newer release of xorg-server-1.20.14, but the affected people haven't tested it (yet), so I can't be confident that it didn't backport the regression. (There is some modesetting change.) You can try https://github.com/vcunat/nixpkgs/commits/p/xorg-server_bump

Can confirm reverting to xorg-server-1.20.14 is solving the issue for me.

**Comment by SuperSandro2000** [comment]
With the amounts of rebuilds we should target staging-next at least.

**Comment by vcunat** [comment]
I assume .14 will be preferable to .13, so I added that here now, but more testing would still be nice.

**Comment by FPtje** [comment]
I just switched to this branch, and I can confirm that this still works, whereas the latest `nixos-unstable` still triggers the issue for me. Thanks!

**Comment by vcunat** [comment]
I think we have enough feedback to stage this at least for unstable.

**Comment by nakoo** [comment]
> I think we have enough feedback to stage this at least for unstable.

I am suffering this issue too. Can I know how to apply this PR to `configuration.nix`? I'm new into NixOS and it's hard to know what I have to change.

I did

```bash
...
let 
  staging = import
    (builtins.fetchTarball https://github.com/nixos/nixpkgs/tarball/staging-next)
    { config.allowUnfree = true; };
in
```

But I'm not sure what I have to change the next since I enabled xserver by `services.xserver.enable = true;`. I added `staging.xorg.xorgserver` into `environment.systemPackages` but Nix didn't build the correct version.

**Comment by vcunat** [comment]
It's relatively easy to use a different nixpkgs version (for everything), e.g.:
```sh
$ nixos-rebuild -I nixpkgs=https://github.com/nixos/nixpkgs/archive/staging-next.tar.gz switch
```

**Comment by nakoo** [comment]
> It's relatively easy to use a different nixpkgs version (for everything), e.g.:
> 
> ```shell
> $ nixos-rebuild -I nixpkgs=https://github.com/nixos/nixpkgs/archive/staging-next.tar.gz switch
> ```

Thanks for your answer. Is there a good way to apply only for the related xserver package? I found it takes a lot of time to build the packages without cache.

**Comment by vcunat** [comment]
The very easiest way I know is to check out nixpkgs of whatever version you want, cherry-pick this pair of commits, and use the above syntax with a local directory instead of URL.

Almost all packages should be in cache: https://hydra.nixos.org/jobset/nixpkgs/staging-next

File: pkgs/servers/x11/xorg/tarballs.list [hunk 1]
```diff
@@ -218,4 +218,4 @@ mirror://xorg/individual/util/lndir-1.0.3.tar.bz2
 mirror://xorg/individual/util/makedepend-1.0.6.tar.bz2
 mirror://xorg/individual/util/util-macros-1.19.3.tar.bz2
 mirror://xorg/individual/util/xorg-cf-files-1.0.7.tar.bz2
-mirror://xorg/individual/xserver/xorg-server-21.1.3.tar.xz
+mirror://xorg/individual/xserver/xorg-server-1.20.13.tar.xz
```
Comment (Artturin) [inline pkgs/servers/x11/xorg/tarballs.list:221]: you forgot to change this to .14, fix in https://github.com/NixOS/nixpkgs/pull/199912

**Pull request merged**